### PR TITLE
NEW elasticsearch backend

### PR DESCRIPTION
**Pull request by snelis**

### CODE REVIEW

File: time_execution/backends/elasticsearch.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        self._setup_index()
+        self._setup_mapping()
+
+    def get_index_with_date(self):
+        return "{}-{}".format(self.index, datetime.now().strftime('%Y.%m.%d'))
+
+    def _setup_index(self):
+        return self.client.indices.create(self.index, ignore=400)
+
+    def _setup_mapping(self):
```
Comment (snelis) [inline time_execution/backends/elasticsearch.py]: I think i will add another PR to make this step optional

Comment (snelis) [inline time_execution/backends/elasticsearch.py]: and configurable / hackable

File: tests/test_hooks.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            return dict(test_key='test value')
+
+        configure(hooks=[test_args, test_metadata])
+
+        go()
+
+    def test_hook_exception(self):
+
+        def exception_hook(exception, **kwargs):
+            self.assertTrue(exception)
```
Comment (quiquepaz) [inline tests/test_hooks.py]: `self.assertInstanceOf(...` would be more meaningful here

File: tests/test_hooks.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import unittest
```
Comment (quiquepaz) [inline tests/test_hooks.py]: Nicer way of testing hooks is actually creating a custom backend with the assertions

**Comment by quiquepaz** [comment]
LGTM

**Pull request merged**